codec/src/test/java/io/netty/handler/codec/compression/Lz4FrameEncoderTest.java

```diff
@@ -52,4 +82,128 @@ protected ByteBuf decompress(ByteBuf compressed, int originalLength) throws Exce
 
         return Unpooled.wrappedBuffer(decompressed);
     }
+
+    @Test
+    public void testAllocateDirectBuffer() {
+        final int blockSize = 100;
+        testAllocateBuffer(blockSize, blockSize - 13, true);
+        testAllocateBuffer(blockSize, blockSize * 5, true);
+        testAllocateBuffer(blockSize, NONALLOCATABLE_SIZE, true);
+    }
+
+    @Test
+    public void testAllocateHeapBuffer() {
+        final int blockSize = 100;
+        testAllocateBuffer(blockSize, blockSize - 13, false);
+        testAllocateBuffer(blockSize, blockSize * 5, false);
+        testAllocateBuffer(blockSize, NONALLOCATABLE_SIZE, false);
+    }
+
+    private void testAllocateBuffer(int blockSize, int bufSize, boolean isDirect) {
+        // allocate the input buffer to an arbitrary size less than the blockSize
+        ByteBuf in = ByteBufAllocator.DEFAULT.buffer(bufSize, bufSize);
+        in.writerIndex(in.capacity());
+
+        ByteBuf out = null;
+        try {
+            Lz4FrameEncoder encoder = newEncoder(blockSize, Lz4FrameEncoder.DEFAULT_MAX_ENCODE_SIZE);
+            out = encoder.allocateBuffer(ctx, in, isDirect);
```

Comment:
Tests are good, but I generally prefer to validate the interface contracts as opposed to internal implementation details (which may be subject to change).